ts/parsers/paragraphparser.ts

```diff
@@ -1,37 +1,126 @@
-import { getValueAtPath } from "../helpers";
+import { checkPath, getValueAtPath } from "../helpers";
 import { ColorParser } from "./";
 
 import {
     PowerpointElement,
     TextAlignment,
     FontAttributes,
     Paragraph,
-    Content
+    Content,
+    List,
+    ListType
 } from "airppt-models-plus/pptelement";
 
 /**
  * Parse the paragraph elements
  */
 export default class ParagraphParser {
+    public static isList(paragraph): boolean {
+        return (
+            checkPath(paragraph, '["a:pPr"][0]["a:buAutoNum"]') ||
+            checkPath(paragraph, '["a:pPr"][0]["a:buChar"]')
+        );
+    }
+
+    public static getParagraph(paragraph): Paragraph {
+        const textElements = paragraph["a:r"] || [];
+        const content = textElements.map((txtElement) => {
+            return {
+                text: txtElement["a:t"] || "",
+                textCharacterProperties: this.determineTextProperties(
+                    getValueAtPath(txtElement, '["a:rPr"][0]')
+                )
+            };
+        });
+
+        return {
+            content: content,
+            paragraphProperties: this.determineParagraphProperties(paragraph)
+        };
+    }
+
+    public static getListlevel(paragraph): number {
+        const level = getValueAtPath(paragraph, '["a:pPr"][0]["$"]["lvl"]');
+
+        return level ? parseInt(level) : 0;
+    }
+
+    public static getListType(paragraph): ListType {
+        if (checkPath(paragraph, '["a:pPr"][0]["a:buAutoNum"]')) {
+            return ListType.Ordered;
+        } else { //getValueAtPath(paragraph, '["a:pPr"][0]["a:buChar"]')
+            return ListType.UnOrdered;
+        }
+    }
+
     public static extractParagraphElements(paragraphs: any[]): PowerpointElement["paragraph"] {
         if (!paragraphs || paragraphs.length === 0) {
             return null;
         }
 
-        return paragraphs.map((paragraph) => {
-            const textElements = paragraph["a:r"] || [];
-            const content = textElements.map((txtElement) => {
-                return {
-                    text: txtElement["a:t"] || "",
-                    textCharacterProperties: this.determineTextProperties(getValueAtPath(txtElement, '["a:rPr"][0]'))
-                };
-            });
+        const paras = [];
+        const stack = [];
+        const paragraph: Paragraph = {
+            list: {
+                listType: ListType.Ordered,
+                listItems: []
+            }
+        };
+        let currentParagraph = paragraph;
+        let currentLevel = -1;
```

Comment:
what is the significance of -1? I thought levels started at 0?

Reply:
Update, looked at the next few lines and I think I know why but please still explain so I get the full picture

Reply:
-1 means that its not the list and its just a paragraph, when we go through paragraphs, and we see a list item we set our first list parent and set the current level at 0

Reply:
OK yes that's what I understood as well, thanks

Reply:
btw, can we start with a null instead or does it absolutely need to be an integer value?

Reply:
needs to be integer, as we decrement increment this value.

Reply:
got it!